### DISCUSSION TITLE
Stuck in Loading after clicking on manage subscriptions

### DISCUSSION THREAD

**Discussion by HMDagher** [maartenpaauw/filament-cashier-billing-provider#23] [Q&A]
installed Stripe and your package 
Business model: use Billable;
AppServiceProvider boot method: Cashier::useCustomerModel(Business::class); 
Tanent Panel (AppPanelProvider in my case): ->tenantBillingProvider(new BillingProvider('default'))
bootstrap/app.php: $middleware->validateCsrfTokens(except: ['stripe/*',]);
published Stripe config, added the plans Array to it and changed the default to match the same name of my Strip product 
altered the migrations to be on Business table 
created 2 products in Stripe as subscriptions
.env : 
STRIPE_KEY=
STRIPE_SECRET=
STRIPE_WEBHOOK_SECRET=
CASHIER_STRIPE_SUBSCRIPTION_DEFAULT_PRICE_ID=

what im doing wrong ? any help 
i checked the console i found CORS error 

https://billing.stripe.com/p/session/test_YWNjdF8xUFJLVWNLdTJ3akpFUGptLF9RbGE4RGdlTzMwQ1dlSnBzdVpFMWxLcVVGT2tiN0ZZ0100DVMpx7tC' (redirected from 'https://domain/app/helozzz/billing') from origin 'https://domain' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled

**Comment by HMDagher** [comment]
and i added webhook to stripe ( domain/stripe/webhook), configured the needed events and added the webhook key to .env 
and when i run stripe:webhook i still get the this msg:
The Stripe webhook was created successfully. Retrieve the webhook secret in your Stripe dashboard and define it as an environment variable.

**Comment by maartenpaauw** [comment]
Did you add the following line to your AppPanelProvider?

`->(Business::class)`

If this line is already added, does your cashier installation work besides filament?

**Reply by HMDagher** [reply]
hey, it's running now, the problem was due to ->SPA on the AppPanelProvider, 
but there's no subscriptions showing :/, i don't know if I'm missing something
products created as subscriptions, configured the customer portal and i made sure i added the plans. 
 
![image](https://github.com/user-attachments/assets/9c9ad18d-eba0-44b6-ab39-634b06d9a65a)

**Reply by maartenpaauw** [reply]
This seems to be a duplicate of #22. And it seems like you ended up on the billing portal page. Did you went there via; Tenant menu -> manage subscription? If so, make sure you are using `->requiresTenantSubscription()` so the tenant is redirected to a Stripe checkout page when they have no subscription (like the tenant within your screenshot).
